### DISCUSSION TITLE
Changing Package source code & rebuilding?

### DISCUSSION THREAD

**Discussion by Dariusz1989** [microsoft/vcpkg#26323] [Q&A] [answered]
Hey

I need to do a small hotfix to one of the packages.

How can I tell vcpkg to recompile "just this one" package? It's the top package that no other package depends on.

Regard
Dariusz

**Comment by Osyotr** [comment, marked as answer]
In classic mode just remove the package and install it again.
In manifest mode this is handled automatically.

**Reply by EduardoJTR** [reply]
Hello, I want to do a similar thing but I don't think your solution will work, Im using box2d and I've made a change to the b2_common.h file, but this change seems to not be working, I think it's beacuse I have to rebuild box2d with this change I've made, how would I do that?

**Reply by dg0yt** [reply]
At which location did you make the change?
For a more permanent solution you will have to create a patch file for the package sources and add it to the port.
For a quick test, you can install (most) ports with `--editable`. This avoid cleaning the source before the build, so that you can edit the source dir without immediately loosing you changes.

**Reply by EduardoJTR** [reply]
I've changed the macro `b2_maxTranslation` from `C:\DEV\vcpkg\installed\x64-windows\include\box2d\b2_common.h`, I didn't really got what you said, what exacly should I do to make this change have effect on my project?

**Reply by dg0yt** [reply]
So you changed the installed file, not the source file. I would expect this to have immediate effect within a certain scope. But it is unclear what you really did and what you really want to achieve. Maybe too much detail for a sub-discussion of a discussion of the original topic.

**Reply by EduardoJTR** [reply]
Well, it's ok, at this point I already gave up.
